src/com/google/enterprise/adaptor/UrlValidator.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.enterprise.adaptor;
```

Comment:
I don't have much of an opinion about a stand-alone class vs an AbstractAdaptor or Context method, but if it's a separate class, maybe a separate package? Could be work looking at some of the other classes we've been thinking of sharing in the library for a theme (beyond, say, "util").